browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/BraveDropdownItemViewInfoListBuilder.java

```diff
@@ -0,0 +1,183 @@
+    @Px
+    private static final int DROPDOWN_HEIGHT_UNKNOWN = -1;
+    private static final int DEFAULT_SIZE_OF_VISIBLE_GROUP = 5;
+
+    @Px
+    private int mDropdownHeight;
+    @Px
+    private int mCalculatedSuggestionsHeight;
+    private Context mContext;
+
+    BraveDropdownItemViewInfoListBuilder(@NonNull Supplier<Tab> tabSupplier,
+            BookmarkState bookmarkState, @NonNull ExploreIconProvider exploreIconProvider,
+            @NonNull OmniboxPedalDelegate omniboxPedalDelegate) {
+        super(tabSupplier, bookmarkState, exploreIconProvider, omniboxPedalDelegate);
+        mActivityTabSupplier = tabSupplier;
+    }
+
+    @Override
+    void initDefaultProcessors(Context context, SuggestionHost host, AutocompleteDelegate delegate,
+            UrlBarEditingTextStateProvider textProvider) {
+        mContext = context;
+        mUrlBarEditingTextProvider = textProvider;
+        super.initDefaultProcessors(context, host, delegate, textProvider);
+        if (host instanceof BraveSuggestionHost) {
+            mBraveSearchBannerProcessor = new BraveSearchBannerProcessor(
+                    context, (BraveSuggestionHost) host, textProvider, delegate);
+        }
+    }
+
+    @Override
+    void onUrlFocusChange(boolean hasFocus) {
+        super.onUrlFocusChange(hasFocus);
+        mBraveSearchBannerProcessor.onUrlFocusChange(hasFocus);
+    }
+
+    @Override
+    void onNativeInitialized() {
+        super.onNativeInitialized();
+        mBraveSearchBannerProcessor.onNativeInitialized();
+    }
+
+    @Override
+    int getVisibleSuggestionsCount(AutocompleteResult autocompleteResult) {
+        // For cases where we don't know how many suggestions can fit in the visile screen area,
+        // make an assumption regarding the group size.
+        if (mDropdownHeight == DROPDOWN_HEIGHT_UNKNOWN) {
+            return Math.min(
+                    autocompleteResult.getSuggestionsList().size(), DEFAULT_SIZE_OF_VISIBLE_GROUP);
+        }
+
+        final List<AutocompleteMatch> suggestions = autocompleteResult.getSuggestionsList();
+
+        @Px
+        int calculatedSuggestionsHeight = 0;
+        int lastVisibleIndex;
+        for (lastVisibleIndex = 0; lastVisibleIndex < suggestions.size(); lastVisibleIndex++) {
+            if (calculatedSuggestionsHeight >= mDropdownHeight) break;
+
+            final AutocompleteMatch suggestion = suggestions.get(lastVisibleIndex);
+            // We do not include suggestions with headers in partial grouping, so terminate early.
+            if (suggestion.getGroupId() != AutocompleteMatch.INVALID_GROUP) {
+                break;
+            }
+
+            final SuggestionProcessor processor =
+                    getProcessorForSuggestion(suggestion, lastVisibleIndex);
+
+            calculatedSuggestionsHeight += processor.getMinimumViewHeight();
+        }
+
+        mCalculatedSuggestionsHeight = calculatedSuggestionsHeight;
```

Comment:
Once again seems to the only change we need, the rest is the copy of the parent. I would rather create a function `getCalculatedSuggestionsHeight` that would only copy relevant part of code.